compiler/src/main/scala/org/scalajs/nscplugin/PrepJSExports.scala

```diff
@@ -441,55 +441,40 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
     }
   }
 
-  /** generate an exporter for a DefDef including default parameter methods */
-  private def genExportDefs(defSym: Symbol, jsName: String, pos: Position) = {
-    val clsSym = defSym.owner
-    val scalaName =
-      jsInterop.scalaExportName(jsName, jsInterop.isJSProperty(defSym))
+  /** generate an exporter for a target including default parameter methods */
+  private def genExportDefs(sym: Symbol, jsName: String, pos: Position) = {
+    val clsSym =
+      if (sym.isConstructor) sym.owner.owner
+      else sym.owner
+
+    val isProperty = sym.isModuleClass || isJSAny(sym) || jsInterop.isJSProperty(sym)
+    val scalaName = jsInterop.scalaExportName(jsName, isProperty)
 
     // Create symbol for new method
-    val expSym = defSym.cloneSymbol
-
-    // Set position of symbol
-    expSym.pos = pos
```

Comment:
We should still have this.

Reply:
That's now part of the `clsSym.newMethod` call (or does the position there not do what I'd expect)?

Reply:
Ah yes, that should indeed do it.